src/item.cpp

```diff
@@ -574,6 +574,16 @@ bool item::stacks_with( const item &rhs, bool check_components ) const
     if( item_vars != rhs.item_vars ) {
         return false;
     }
+    // check if components are the same, otherwise don't stack
+    if( components.size() != rhs.components.size() ) {
+        return false;
+    } else {
+        for( int i = 0; components.size(); i++ ) {
```

Comment:
I'm getting a crash here when an item that has components is compared with another item. Since components.size() always returns a value over 0 it will keep looping until it goes out of bounds and causes an exception.

Furthermore this bit of code duplicates the logic I added [here](https://github.com/CleverRaven/Cataclysm-DDA/pull/27392/files#diff-f7ca61cf749eba039da961e8a7be1630R600) that handles component comparison by using a flag passed to item::stacks_width(). This can be solved by either removing this block, updating the branch and using my component comparison by calling item::stacks_width(item, true) or fixing it here and removing the changes I did. I recommend keeping the flag and passing the boolean value to the function so items can still be stacked in inventory screens and only un-stacked when needed (i.e. when showing the disassembly window).